lib/paper_trail/model_config.rb

```diff
@@ -150,17 +142,17 @@ def setup_versions_association(klass)
       klass.has_many(
         klass.versions_association_name,
         lambda do
-          relation = order(model.timestamp_sort_order)
-          item_type = klass.paper_trail.send(:versions_association_item_type)
-          relation = relation.unscope(where: :item_type).where(item_type: item_type)
-          relation
+          order!(model.timestamp_sort_order)
+          unless klass.descendants.any? && klass.columns.exclude?(klass.inheritance_column)
+            unscope(where: :item_type).where(item_type: klass.name)
+          end
```

Comment:
I don't fully understand this `unless` condition and I could use an explanation comment, pretty please.